pkg/gw/immugw_server_mock.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2019-2020 vChain, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gw
```

Comment:
If Im not wrong you could  move this mock inside  gw/gwtest folder in a gwtest package and should work. 
```package gwtest

import (
"github.com/codenotary/immudb/pkg/api/schema"
	"github.com/codenotary/immudb/pkg/gw"
	"github.com/codenotary/immudb/pkg/logger"
)

// ImmuGwServerMock ...
type ImmuGwServerMock struct {}


// Start ...
func (igm ImmuGwServerMock) Start() error {
	return nil
}

// Stop ...
func (igm ImmuGwServerMock) Stop() error {
	return nil
}

// WithClient ...
func (igm ImmuGwServerMock) WithClient(client schema.ImmuServiceClient) gw.ImmuGw {
	return igm
}

// WithLogger ...
func (igm ImmuGwServerMock) WithLogger(logger logger.Logger) gw.ImmuGw {

	return igm
}

// WithOptions ...
func (igm ImmuGwServerMock) WithOptions(options gw.Options) gw.ImmuGw {
	return igm
}
```